libs/@local/harpc/client/typescript/src/net/Connection.ts

```diff
@@ -180,28 +176,34 @@ const task = (connection: ConnectionImpl) =>
 export const makeUnchecked = (
   transport: internalTransport.Transport,
   config: ConnectionConfig,
-  peer: PeerId | Multiaddr | Multiaddr[],
+  peer: Transport.Address,
 ) =>
   Effect.gen(function* () {
-    const connection = yield* Effect.tryPromise((abort) =>
-      transport.dial(peer, { signal: abort }),
-    );
+    const connection = yield* internalTransport.connect(transport, peer);
 
     const stream = yield* Effect.acquireRelease(
-      Effect.tryPromise((abort) =>
-        connection.newStream("/harpc/1.0.0", {
-          signal: abort,
-          maxOutboundStreams: config.maxOutboundStreams,
-          runOnLimitedConnection: config.runOnLimitedConnection,
-        }),
-      ),
-      (_) => Effect.promise(() => _.close()),
+      Effect.tryPromise({
+        try: (abort) => {
+          const options: NewStreamOptions = { signal: abort };
+
+          if (config.maxOutboundStreams) {
+            options.maxOutboundStreams = config.maxOutboundStreams;
+          }
+          if (config.runOnLimitedConnection) {
```

Comment:
The default is `false`, but it's also an oversight on my part! I can actually rollback most of this now, I needed to break this out because I was trying to figure out the eslint problem and enabled `exactOptionalPropertyTypes`.

Fixed in [a3d33f4](https://github.com/hashintel/hash/pull/5836/commits/a3d33f434f4bc7267042871b4adde3b9271818d0) and copied documentation from libp2pjs about the two options.